src/dependency-manager/src/utils/interpolation/interpolation-context.ts

```diff
@@ -0,0 +1,38 @@
+import { ParameterValueV2 } from '../../service-config/base';
+
+export type EnvironmentParameterMap = { [key: string]: ParameterContext };
+
+export interface InterpolationContext {
```

Comment:
Oh here it is again. Same q. Would prefer not to duplicate the service config.

Reply:
This object is not the same shape as a ServiceConfig which is why we need it. The interfaces block is enriched for example.

If we made the interpolation context the ServiceConfig we'd either have to:
(1) remove support for the enriched interface expressions `*.interfaces.external.*`
(2) change the ServiceConfig so that we can write those expressions.

The problem with the latter is that the ServiceConfig represents the shape of the file that the user is allowed to give us. That's what we validate and that's what we version. This is different from the shape of the object that we want to enrich.

I'd be open to using the actual `node_config` as the interpolation context, but this follows the same lines of what I've been saying: the `node_config` is not a `ServiceConfig` if we're going to write to it and enrich it. It deserves its own type.

Reply:
I'm leaning towards option 2 since the ServiceConfig is the transformed object not the shape of the file. However, I think that should be another story that can follow this one. I can live with InterpolationContext for some period of time.